modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/RelationshipObjectFieldBusinessType.java

```diff
@@ -87,83 +90,72 @@ public Object getValue(
 			ObjectField objectField, long userId, Map<String, Object> values)
 		throws PortalException {
 
-		if (!Objects.equals(
+		if (Objects.equals(
 				objectField.getRelationshipType(),
-				ObjectRelationshipConstants.TYPE_ONE_TO_MANY) ||
-			values.containsKey(objectField.getName())) {
-
-			Object value = values.get(objectField.getName());
-
-			Long valueLong = GetterUtil.getLong(value);
-
-			if (Validator.isNull(valueLong)) {
-				return value;
-			}
-
-			ObjectRelationship objectRelationship =
-				_objectRelationshipLocalService.
-					fetchObjectRelationshipByObjectFieldId2(
-						objectField.getObjectFieldId());
-
-			ObjectDefinition objectDefinition =
-				_objectDefinitionLocalService.getObjectDefinition(
-					objectRelationship.getObjectDefinitionId1());
-
-			if (objectDefinition.isUnmodifiableSystemObject()) {
-				SystemObjectDefinitionManager systemObjectDefinitionManager =
-					_systemObjectDefinitionManagerRegistry.
-						getSystemObjectDefinitionManager(
-							objectDefinition.getName());
-
-				BaseModel<?> baseModel =
-					systemObjectDefinitionManager.
-						getBaseModelByExternalReferenceCode(
-							systemObjectDefinitionManager.
-								getBaseModelExternalReferenceCode(valueLong),
-							objectDefinition.getCompanyId());
-
-				return baseModel.getPrimaryKeyObj();
-			}
-
-			ObjectEntry objectEntry = _objectEntryLocalService.getObjectEntry(
-				valueLong);
-
-			if (!Objects.equals(
-					objectDefinition.getObjectDefinitionId(),
-					objectEntry.getObjectDefinitionId())) {
-
-				throw new ObjectEntryValuesException.InvalidValue(
-					objectField.getName());
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY)) {
+
+			String objectRelationshipERCObjectFieldName =
+				ObjectFieldSettingUtil.getValue(
+					ObjectFieldSettingConstants.
+						NAME_OBJECT_RELATIONSHIP_ERC_OBJECT_FIELD_NAME,
+					objectField);
+
+			if (values.containsKey(objectRelationshipERCObjectFieldName)) {
+				ObjectDefinition objectDefinition = _getObjectDefinition(
+					objectField);
+
+				String externalReferenceCode = GetterUtil.getString(
+					values.get(objectRelationshipERCObjectFieldName));
+
+				if (objectDefinition.isUnmodifiableSystemObject()) {
+					SystemObjectDefinitionManager
+						systemObjectDefinitionManager =
+							_systemObjectDefinitionManagerRegistry.
+								getSystemObjectDefinitionManager(
+									objectDefinition.getName());
+
+					BaseModel<?> baseModel =
+						systemObjectDefinitionManager.
+							getBaseModelByExternalReferenceCode(
+								externalReferenceCode,
+								objectDefinition.getCompanyId());
+
+					return baseModel.getPrimaryKeyObj();
```

Comment:
@p-albuquerque you can create a private method to reuse code:

```
private Object _getPrimaryKeyObj(
		String externalReferenceCode, ObjectDefinition objectDefinition,
		Long primaryKey)
	throws PortalException {

	SystemObjectDefinitionManager systemObjectDefinitionManager =
		_systemObjectDefinitionManagerRegistry.
			getSystemObjectDefinitionManager(objectDefinition.getName());

	if (externalReferenceCode == null) {
		externalReferenceCode =
			systemObjectDefinitionManager.getBaseModelExternalReferenceCode(
				primaryKey);
	}

	BaseModel<?> baseModel =
		systemObjectDefinitionManager.getBaseModelByExternalReferenceCode(
			externalReferenceCode, objectDefinition.getCompanyId());

	return baseModel.getPrimaryKeyObj();
}
```

Reply:
Got it!

Reply:
Actually, I can improve it